streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -931,6 +932,22 @@ public synchronized boolean close(final long timeout, final TimeUnit timeUnit) {
         return close(timeoutMs);
     }
 
+    /**
+     * Attempts to shutdown an application.
+     * If there is an alive StreamThread it will succeed, if there is not it will fail
+     *
+     * @return Will return true if shutdown is initiated false if it is not possible.
+     */
+    public boolean shutdownApplication() {
```

Comment:
This is the idea for a shutdown that users can initiate without throwing an exception. It will still need at least one alive stream thread